blaze/expr/arithmetic.py

```diff
@@ -220,7 +221,7 @@ def scalar_coerce(_, val):
 
 @dispatch(ct.DateTime, _strtypes)
 def scalar_coerce(_, val):
-    return dt_parse(val)
+    return pd.Timestamp(val)
```

Comment:
It handles them just fine, except when rendering literal value. You can use `literal_column(dt.strftime('...'))` or it's up to the dialect to manage literal rendering. You can of course pass `datetime` around in expressions and it deals with sending those values to the database quite nicely.